src/compiler/parser.rs

```diff
@@ -387,7 +386,7 @@ impl ParserType {
             let operator_token = self.parse_token_node();
             let right = self.parse_binary_expression_or_higher(new_precedence);
             left_operand = self
-                .make_binary_expression(left_operand, operator_token, right)
+                .make_binary_expression(left_operand, operator_token.into(), right)
```

Comment:
Here this is calling `BaseNode::into::<Node>()` (which wraps in a `Node::BaseNode`)...

Reply:
If you wanted to get fancy, you could make `make_binary_expression` have a type parameter of something convertible into a `Node`, and then you wouldn't need `into` at the call sites.

Reply:
Sure, updated to accept a `<TNode: Into<Node>>`